### ISSUE TITLE
Timeout issues

### ISSUE THREAD

**Issue by bananamufu** [scottdurow/dataverse-gen#9]
Hi,

I'm getting timeouts errors when I try to generate the metadata using npx dataverse-gen. If I reduce the fetched entities to like below 10, it goes through sometimes, however sometimes it times out even on only one entity fetched.  The entity it fails on is different basically every time.

Any help would be appreciated.

Kind regards,
Tom

```Fetching EDMX metadata
Fetching CDS metadata for account
Fetching CDS metadata for activityparty
Fetching CDS metadata for activitypointer
Fetching CDS metadata for annotation

Error:Exception in execute:connect ETIMEDOUT ip:443
Stack:Error: connect ETIMEDOUT ip:443
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1141:16)
{"errno":"ETIMEDOUT","code":"ETIMEDOUT","syscall":"connect","address":"ip","port":443}```

**Comment by sefarov92** [comment]
Hi, any update on this? Thank you

**Comment by scottdurow** [comment]
When you get the timeout - how long does the fetching metadata step wait approximately before you see the error? Is there anything else happening on the server at that time such as importing solutions?

**Comment by sefarov92** [comment]
It happens when fetching more than 10 entities, immediately throws error while fetching CDS Metadata - Error:Exception in execute:connect ETIMEDOUT  Stack:Error: connect ETIMEDOUT at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1157:16) {"errno":-4039,"code":"ETIMEDOUT","syscall":"connect","address":"","port":443} and there was no importing or publishing activity was running on the instance.

**Comment by scottdurow** [comment]
Thanks for the info - so this looks like a different error to the op issue - does it consistently happen every time?

**Comment by sefarov92** [comment]
Yeah so far, it keeps happening even with different instance or machine, when we try to get more than 10 entities metadata.

**Comment by scottdurow** [comment]
What OS are you using - and what version of node?
It seems odd that it is so consistent for 10 entities - and it times out immediately rather than waiting for the default OS timeout before terminating.

**Comment by sefarov92** [comment]
Sorry took some time to reply. It's not consistently breaking on 10 entities, it just breaks when we supply more than 10 entities, e.g., if we supply 20 entity names in dataverse-gen.json. It will crash while retrieving the 15th entity metadata, and if we re-run this, it will crash again while retrieving the 4th entity metadata, but it will never completely retrieve those 20 entity metadata, no matter how many times we try. Tested with Win10 latest build, Node 14.18.3, and 16.14.2.
